pkg/sql/opt/cat/catalog.go

```diff
@@ -83,6 +83,14 @@ type Flags struct {
 // returned by the Resolve methods (schemas and data sources) *must* be
 // immutable after construction, and therefore also thread-safe.
 type Catalog interface {
+	// ResolveDatabase locates a database with the given name and returns it.
+	// If no name is provided, it will return the current database.
+	// An error is returned if no database with the given name exists or in the
+	// case of an empty name, there is no current database.
+	// TODO(yang): This function can be extended if needed in the future
+	// to return a new cat.Database type similar to cat.Schema.
+	ResolveDatabase(ctx context.Context, flags Flags, name string) (tree.Name, error)
```

Comment:
nit: I think this more a `LookupDatabaseName` than resolving :)
my 2c is that we can just call it `DatabaseExists` for now... Your call.

Reply:
Changed to `LookupDatabaseName`! I think `DatabaseExists` would imply it just returns a `bool` or `error`, which I also considered but it wouldn't work in this case because we want it to work for the empty string (i.e return current database name) too.